### PR TITLE
Update spin dependency

### PR DESCRIPTION
**Pull request by estebangarcia**
Needed to update spin dependency for supporting LDAP authentication (https://github.com/spinnaker/spin/pull/188)

This PR depends on https://github.com/armory-io/terraform-provider-spinnaker/pull/23 as it will fail because of the lack of the 'default-headers' flag on gate client creation.

### CODE REVIEW

File: spinnaker/provider.go [hunk 1]
```diff
@@ -1,19 +1,21 @@
 package spinnaker
 
 import (
+	"log"
+
+	spin_config "github.com/estebangarcia/spin/config"
```
Comment (nilium) [inline spinnaker/provider.go]: It might be better to call this spinconfig. Underscores in Go are pretty unusual.

File: spinnaker/provider.go [hunk 2]
```diff
@@ -1,19 +1,21 @@
...
 func Provider() *schema.Provider {
 	return &schema.Provider{
 		Schema: map[string]*schema.Schema{
 			"server": {
 				Type:        schema.TypeString,
-				Required:    true,
+				Optional:    true,
 				Description: "URL for Gate",
-				DefaultFunc: schema.EnvDefaultFunc("GATE_URL", nil),
+				DefaultFunc: schema.EnvDefaultFunc("SPIN_GATE_ENDPOINT", nil),
```
Comment (nilium) [inline spinnaker/provider.go]: Changing this may break existing users. The original name is definitely not great since it isn't using PROVIDER_ATTR format, but maybe this change should be a separate PR so that it's easier to accept this one.

File: spinnaker/provider.go [hunk 3]
```diff
@@ -49,17 +51,30 @@ type gateConfig struct {
...
+		cfg, err = spin_config.Parse()
+	}
+
+	cfg.Gate.Insecure = ignoreCertErrors
+
+	if server != "" {
+		cfg.Gate.Endpoint = server
+	}
+
+	log.Printf("%v", cfg)
```
Comment (nilium) [inline spinnaker/provider.go]: Logging line should be removed. It could potentially leak secrets to logs.

File: spinnaker/provider.go [hunk 4]
```diff
@@ -49,17 +51,30 @@ type gateConfig struct {
...
+	var cfg spin_config.Config
+	var err error
+
+	if config != "" {
+		cfg, err = spin_config.ParseFromFile(config)
+	} else {
+		cfg, err = spin_config.Parse()
+	}
+
+	cfg.Gate.Insecure = ignoreCertErrors
```
Comment (nilium) [inline spinnaker/provider.go]: Should this maybe be an `||` to merge the config file and provider values? Or should the config file's insecure option be ignored completely?

File: spinnaker/api/pipeline.go [hunk 5]
```diff
@@ -5,7 +5,7 @@ import (
 	"net/http"
 
 	"github.com/mitchellh/mapstructure"
-	gate "github.com/spinnaker/spin/cmd/gateclient"
```
Comment (nilium) [inline spinnaker/api/pipeline.go]: This should be pointing to the upstream spinnaker/spin, not a fork.

File: spinnaker/api/application.go [hunk 6]
```diff
@@ -6,8 +6,8 @@ import (
 	"strings"
 	"time"
 
+	gate "github.com/estebangarcia/spin/gateclient"
 	"github.com/mitchellh/mapstructure"
-	gate "github.com/spinnaker/spin/cmd/gateclient"
```
Comment (nilium) [inline spinnaker/api/application.go]: Ditto.

File: go.mod [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+	cloud.google.com/go v0.39.0
+	github.com/agext/levenshtein v1.2.2
+	github.com/apparentlymart/go-cidr v1.0.0
+	github.com/apparentlymart/go-textseg v1.0.0
+	github.com/armon/go-radix v1.0.0
+	github.com/aws/aws-sdk-go v1.19.42
+	github.com/bgentry/go-netrc v0.0.0-20140422174119-9fd32a8b3d3d
+	github.com/bgentry/speakeasy v0.1.0
+	github.com/blang/semver v0.0.0-20190414102917-ba2c2ddd8906
+	github.com/estebangarcia/spin v0.0.0
```
Comment (nilium) [inline go.mod]: This should use spinnaker/spin, not a fork. When using v0.0.0, you should also format it as `v0.0.0-${DATE}-${COMMIT_12}`. For the LDAP changes that're upstream, this would be `v0.0.0-20190530150642-535d2dc1b985` for the current master.

File: go.mod [hunk 8]
```diff
@@ -0,0 +1,71 @@
+module github.com/armory-io/terraform-provider-spinnaker
```
Comment (nilium) [inline go.mod]: The README should be updated to document how to build this if Go modules are used. Otherwise, it may be better in a separate pull request, since this project already tracks its dependencies using dep.

**Comment by estebangarcia** [comment]
@nilium apologies for this. I was doing some tests with a forked version of spin I have in my account and I pushed it all to the incorrect branch.

My original change was just updating the dependency not all that you see here now :)

I’m going to close this and reopen with the proper change.

Again I’m sorry.

**Comment by nilium** [comment]
@estebangarcia Absolutely nothing to apologize for! Looking forward to seeing the next PR. I think these changes will be useful.
